src/plugins/visualizations/public/embeddable/visualize_embeddable.ts

```diff
@@ -393,7 +399,14 @@ export class VisualizeEmbeddable extends Embeddable<VisualizeInput, VisualizeOut
     });
   };
 
-  public supportedTriggers() {
-    return this.vis.type.getSupportedTriggers?.() ?? [];
+  public supportedTriggers(): TriggerId[] {
+    const triggers = this.vis.type.getSupportedTriggers?.() ?? [];
+    // TODO: this is a flaw of current "dynamic" nested triggers implementation
```

Comment:
is this something we'll be able to address and is there a plan for it ?

Reply:
@ppisljar, No, I don't have a plan. This came up during implementation. I'll create an issue where we can discuss how to handle this nicer.